backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/quickinput/ddorderline/DDOrderLineQuickInputDescriptorFactory.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * #%L
+ * metasfresh-webui-api
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.ui.web.quickinput.ddorderline;
+
+import com.google.common.collect.ImmutableSet;
+import de.metas.i18n.IMsgBL;
+import de.metas.lang.SOTrx;
+import de.metas.ui.web.quickinput.IQuickInputDescriptorFactory;
+import de.metas.ui.web.quickinput.QuickInputConstants;
+import de.metas.ui.web.quickinput.QuickInputDescriptor;
+import de.metas.ui.web.quickinput.QuickInputLayoutDescriptor;
+import de.metas.ui.web.window.datatypes.DocumentId;
+import de.metas.ui.web.window.datatypes.DocumentType;
+import de.metas.ui.web.window.datatypes.LookupValue;
+import de.metas.ui.web.window.descriptor.DetailId;
+import de.metas.ui.web.window.descriptor.DocumentEntityDescriptor;
+import de.metas.ui.web.window.descriptor.DocumentFieldDescriptor;
+import de.metas.ui.web.window.descriptor.DocumentFieldWidgetType;
+import de.metas.ui.web.window.descriptor.sql.SqlLookupDescriptor;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.expression.api.ConstantLogicExpression;
+import org.compiere.util.DisplayType;
+import org.eevolution.model.I_DD_OrderLine;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.Set;
+
+@Component
+public class DDOrderLineQuickInputDescriptorFactory implements IQuickInputDescriptorFactory
+{
+	private final IMsgBL msgBL = Services.get(IMsgBL.class);
+
+	private final DDOrderLineQuickInputCallout ddOrderLineQuickInputCallout;
+
+	private static final int M_PRODUCT_STOCKED_AD_REFERENCE_ID = 171;
+	private static final int DD_OrderLine_M_HU_PI_Item_Product = 540299;
```

Comment:
use `HUPIItemProductId` instead of `int`.

Reply:
also can we use a sysconfig instead of hardconding it?

Reply:
that's not a HUPIItemProductId, is an ad_val_rule_id, I know, bad naming 🤦